activesupport/lib/active_support/cache.rb

```diff
@@ -763,6 +772,40 @@ def dump(entry)
       end
     end
 
+    MARSHAL_PREFIX =  "\x04\b".b.freeze
```

Comment:
I see it hasn't changed in nearly 20 years, but is Marshal's version officially frozen?

Reply:
I don't think it's officially frozen, however it's officially documented and it's unlikely to change any time soon. If the legacy format is to be deprecated in a couple Rails version, I don't think it's that risky to rely on it.